lib/components/user/saved-trip-screen.js

```diff
@@ -102,12 +103,35 @@ class SavedTripScreen extends Component {
 
   _handleSaveNewTrip = async () => {
     await this._updateMonitoredTrip()
-    this._goToTripPlanner()
+
+    // For new trips, if a trip with the same ui_activeSearch value in queryParams
+    // appears in the monitored trips redux state,
+    // then we know that save was successful and we can proceed.
+    const { activeSearchId } = this.props
+
+    const savedTrip = this.props.monitoredTrips.find(trip => {
+      const tripQueryParams = qs.parse(trip.queryParams.split('?')[1])
+      return tripQueryParams.ui_activeSearch === activeSearchId
+    })
+
+    if (savedTrip) {
+      this._goToTripPlanner()
+    }
   }
 
   _handleSaveTripEdits = async () => {
+    const { monitoredTrip } = this.state
+
     await this._updateMonitoredTrip()
-    this._goToSavedTrips()
+
+    // For existing trips, if the lastUpdated value (long) for the trip with the same id in monitored trips redux state
+    // is more recent that our copy, then we know that save was successful and we can proceed.
+    // (The state is not updated upon save because the next normal step is to close this screen.)
+    const savedTrip = this.props.monitoredTrips.find(trip => trip.id === monitoredTrip.id)
+
+    if (savedTrip.lastUpdated > monitoredTrip.lastUpdated) {
```

Comment:
See above comment about moving this check (and probably the need for comparing lastUpdated) to the action itself.

Reply:
Good call. It's cleaner now in https://github.com/opentripplanner/otp-react-redux/pull/232/commits/861f5e8d951548dac61c07d5083f2b60682f604a.